web/src/engine/js-processor/src/kbdInterface.ts

```diff
@@ -30,93 +20,9 @@ export class KeyInformation {
   modifiers: number;
 }
 
-/*
-* Type alias definitions to reflect the parameters of the fullContextMatch() callback (KMW 10+).
-* No constructors or methods since keyboards will not utilize the same backing prototype, and
-* property names are shorthanded to promote minification.
-*/
-type PlainKeyboardStore = string;
-
-export type KeyboardStoreElement = (string|StoreNonCharEntry);
-export type ComplexKeyboardStore = KeyboardStoreElement[];
-
-type KeyboardStore = PlainKeyboardStore | ComplexKeyboardStore;
-
-export type VariableStore = {[name: string]: string};
-
-type RuleChar = string;
-
-class RuleDeadkey {
-  /** Discriminant field - 'd' for Deadkey.
-   */
-  t: 'd';
-
-  /**
-   * Value:  the deadkey's ID.
-   */
-  d: number; // For 'd'eadkey; also reflects the Deadkey class's 'd' property.
-}
-
-class ContextAny {
-  /** Discriminant field - 'a' for `any()`.
-   */
-  ['t']: 'a';
-
-  /**
-   * Value:  the store to search.
-   */
-  ['a']: KeyboardStore; // For 'a'ny statement.
-
-  /**
-   * If set to true, negates the 'any'.
-   */
-  ['n']: boolean|0|1;
-}
-
-class RuleIndex {
-  /** Discriminant field - 'i' for `index()`.
-   */
-  ['t']: 'i';
-
-  /**
-   * Value: the Store from which to output
-   */
-  ['i']: KeyboardStore;
-
-  /**
-   * Offset: the offset in context for the corresponding `any()`.
-   */
-  ['o']: number;
-}
-
-class ContextEx {
-  /** Discriminant field - 'c' for `context()`.
-   */
-  ['t']: 'c';
-
-  /**
-   * Value:  The offset into the current rule's context to be matched.
-   */
-  ['c']: number; // For 'c'ontext statement.
-}
-
-class ContextNul {
-  /** Discriminant field - 'n' for `nul`
-   */
-  ['t']: 'n';
-}
-
-class StoreBeep {
-  /** Discriminant field - 'b' for `beep`
-   */
-  ['t']: 'b';
-}
```

Comment:
Why is this section not being moved with the rest of this file?  It relates solely to rule definitions, and that stuff conceptually belongs to `js-processor` in my mind.